.github/workflows/ci.yml

```diff
@@ -73,6 +73,27 @@ jobs:
           command: test
           args: --verbose --target ${{ matrix.target }}
 
+  wasi:
+    name: wasi
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        rust:
+          - stable
+          - beta
+          - nightly
+    env:
+      CARGO_TERM_COLOR: always
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: true    
+      - name: Install WASI
+        run: cargo install cargo-wasi    
+      - name: Run build on wasm32-wasi
+        run: |
+          cargo wasi build
+
```

Comment:
Could you try running `cargo wasi test` as well?

Reply:
"cargo wasi" uses wasitime which allows for compiling web assembly and running the basics of unit tests however it does not implement full virtual machine.

in this case its file system operations are too limited to pass your tests.

```
           5: 0xacf9f - <unknown>!std::panicking::begin_panic::h5e49e2e7ce9ffb12
           6: 0xaebb1 - <unknown>!std::sys::wasi::os::temp_dir::hf34626cb00f6ff15
           7: 0xaef3c - <unknown>!std::env::temp_dir::h79c65cb75118b89f
           8: 0xa3001 - <unknown>!tempfile::Builder::tempdir::h7f25734b07600992
           9: 0xa0919 - <unknown>!tempfile::dir::TempDir::new::hbfdc4237f6b1ba33
```

there are other runtime's you could try such as "wasmer" which is what tokera.sh is using but these are not easy to integrate with CI/CD just yet.

its a bit too much work for me to take on

Reply:
Got it, thanks for trying that